src/shared/install.c

```diff
@@ -2421,11 +2431,11 @@ int unit_file_link(
                 if (r < 0)
                         return install_changes_add(changes, n_changes, r, *file, NULL);
 
-                r = in_search_path(&lp, *file);
+                r = underneath_search_path(&lp, *file);
+                if (r > 0)
+                        r = -ETXTBSY;
                 if (r < 0)
                         return install_changes_add(changes, n_changes, r, *file, NULL);
-                if (r > 0)
-                        continue;
```

Comment:
mhmm, this is a compta break, no?

previously you could call "systemctl link" on any unit file, regardless where it was, and it would be a clean NOP if it was already in the search path. 

I think we should retain that behaviour, it does make sense (and we should remain compatible)

i.e. change the check to say: "if the unit is directly in the search path, make things a NOP; if the unit is somwehere further down, error out".

Reply:
I guess. I'm not 100% convinced that this behaviour is useful in real life, because I think you'd generally "know" if a file is in the search path or not… But keeping backwards-compat is a good argument.